x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferencePlugin.java

```diff
@@ -232,12 +234,30 @@ public Collection<?> createComponents(PluginServices services) {
         }
 
         if (elasticInferenceUrl != null) {
```

Comment:
I think it's because we have introduced a new feature flag and the old one is deprecated. So `elasticInferenceUrl` is artifact of those 2 feature flags logic. It's defined [above](https://github.com/elastic/elasticsearch/pull/116423/files#diff-732d8987f7a9e77342d457a482d8f7d481e86778344499b5fa9b0877c13137ecR224-L232). 

cc @timgrein

Reply:
I saw the logic of `elasticInferenceUrl` and got confused, as I would have expected that feature flag hides everything related to the inference service, including settings.

Reply:
Fair point, looking at this a bit more closely I would also suggest we abstract away the fact that we renamed the flag and the url setting:

There could also be a case, where you use a combination of the old flag and the new url setting and vice versa, which we wouldn't handle correctly I think, right? cc @jaybcee

```
method feature_flag_enabled {
  if old_feature_flag_enabled {
    log.warning(...)
  }
  
  return old_feature_flag_enabled || new_feature_flag_enabled
}

method eis_url {
  if old_url_setting {
    log.warning(...)
    return old_url_setting
  }
  
  return new_url_setting
}

if feature_flag_enabled() {
  var eis_url = eis_url()
  
  //instantiate http client etc...
}
```

Is this what you mean @slobodanadamovic?

Reply:
Yes, you are correct, if the new flag is active, it will only seek the "new" setting, which I agree is confusing. If we are to make any such changes, I think it might make more sense to simply remove the old one. I don't know if this acceptable or not. It's still to be "tested" in QA. WDYT?

Reply:
> If we are to make any such changes, I think it might make more sense to simply remove the old one

Agree on that, if we're sure we've removed all references to both the old feature flag and the old setting everywhere, then I'm 100% for removing it. We can also do this later on with a separate PR though, to make it easily revertible, if we miss some place and stuff breaks.

Reply:
Yeah, it depends on how urgent this "odd" behaviour is deemed to be. If its "urgent", I can take a few minutes and implement what you suggested above. Otherwise, we can ensure that the correct flag is used.

The [feature flag](https://github.com/search?q=eis_feature_flag+org%3Aelastic&type=code) only needs to be changed in `serverless-gitops` (outside of some readme).

The [setting itself](https://github.com/search?q=xpack.inference.eis.gateway.url+org%3Aelastic&type=code) needs to be changed `elastic/cloud*` and is inflight in `elasticsearch-controller`.

What do we prefer?

Reply:
I don't think it's super urgent tbh. Let's create an issue and make sure it's part of M5 or M6, we can address it after mTLS merged, so we're not adding any unnecessary conflicts. Thanks 👍

Reply:
addressed [here](https://github.com/elastic/elasticsearch/pull/116423/commits/846fa32cea255ba70bd515749f1c6c3081b032af). Now we rely on `isElasticInferenceServiceEnabled()`.